src/structures/MessageActionRow.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const BaseMessageComponent = require('./BaseMessageComponent');
+const { MessageComponentTypes } = require('../util/Constants');
+
+/**
+ * Represents an ActionRow containing message components.
+ * @extends {BaseMessageComponent}
+ */
+class MessageActionRow extends BaseMessageComponent {
+  /**
+   * Components that can be placed in a MessageActionRow
+   * * MessageButton
+   * @typedef {MessageButton} MessageActionRowComponent
+   */
+
+  /**
+   * Options for components that can be placed in a MessageActionRow
+   * * MessageButtonOptions
+   * @typedef {MessageButtonOptions} MessageActionRowComponentOptions
+   */
+
+  /**
+   * Data that can be resolved into a components that can be placed in a MessageActionRow
+   * * MessageActionRowComponent
+   * * MessageActionRowComponentOptions
+   * @typedef {MessageActionRowComponent|MessageActionRowComponentOptions} MessageActionRowComponentResolvable
+   */
+
+  /**
+   * @typedef {BaseMessageComponentOptions} MessageActionRowOptions
+   * @property {MessageActionRowComponentResolvable[]} [components]
+   * The components to place in this ActionRow
+   */
+
+  /**
+   * @param {MessageActionRow|MessageActionRowOptions} [data={}] MessageActionRow to clone or raw data
+   */
+  constructor(data = {}) {
+    super({ type: 'ACTION_ROW' });
+
+    this.components = (data.components ?? []).map(c => BaseMessageComponent.create(c, null, true));
```

Comment:
This property currently does not appear in the documentation because it's missing a docstring.